### PR TITLE
Sgp30 sensor improvements

### PR DESCRIPTION
**Pull request by huubeikens**
## Description:

Updated the SGP30 sensor with configuration options to persistently store the eCO2/TVOC baselines so that it survives a power cycle. It removes the need to recompile and OTA the configuration file when the baselines have been determined after the calibration period (of 12 hours). 

As a convenience, the baseline values can also be configured as sensors so that they are published to Home assistant. 

Last, the update interval default is set to 1 second which is advised by the datasheet.

Please note that the amount of writes to the NVS (non-volatile-storage) is kept to a mininum to prevent wear and tear. 

**Related issue (if applicable):** fixes <link to issue>
#673
#668

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#982

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by kbx81** [comment]
I am running this branch on four sensor boards I have around my place -- so far, so good! I'll report back in a few days with an update. 😄

**Comment by huubeikens** [comment]
Thanks for testing!

**Comment by kbx81** [comment]
So far, so good! It's been almost two weeks and still going strong...

**Comment by darkson95** [comment]
Thanks for the cool work! This solves exactly the problems I would like to eliminate. 😎
Without building up pressure, I'd like to know when this PR goes live, as I'm having trouble reading the baseline in time.

**Comment by huubeikens** [comment]
Thanks! I've been testing it for a few months now without any issues. 

@OttoWinter : could you please merge this PR into the master branch? Much thanks in advance.

**Comment by agners** [comment]
There are some linter errors it seems. Otherwise, from my point of view the PR looks good.

File: esphome/components/sgp30/sgp30.h [hunk 1]
```diff
@@ -3,16 +3,25 @@
...
+#include "esphome/core/preferences.h"
 #include <cmath>
 
 namespace esphome {
 namespace sgp30 {
 
+struct SGP30Baselines {
+  uint16_t eco2;
+  uint16_t tvoc;
+} PACKED;  // NOLINT
```
Comment (agners) [inline esphome/components/sgp30/sgp30.h:15]: I think this `NOLINT` is not required here.

Comment (huubeikens) [reply esphome/components/sgp30/sgp30.h]: You're right. I've removed it. I've also compiled it again towards esphome:dev and solved a yield error I got. 
See new commits.
This PR should now be ready for the merge into esphome:dev.

**Review by kbx81 — APPROVED** [review]
This has been running on several of my custom sensor boards for quite a while now and it appears to be stable and is working as expected. Nothing jumps out at me code-wise...probably good to merge.

**Review by agners — APPROVED** [review]

**Comment by huubeikens** [comment]
I see that all checks have passed. Please let me know if I need to do something more to get this change merged.

**Comment by darkson95** [comment]
A few days have now passed since this PR is ready to merge. What is still missing here? Is there anything I can assist with so that the merge is accelerated?

**Review by jesserockz — APPROVED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**